src/adapter/dummyadapter.c

```diff
@@ -77,15 +77,15 @@ static bool DummyAdapter_CreateIdTransaction(DIDAdapter *_adapter, const char *p
     }
 
     DIDDocument *doc = DIDRequest_FromJson(&info->request, root);
-    if (!doc)
+    if (!doc && strcmp(info->request.header.op, "deactivate"))
         goto errorExit;
 
     if (strcmp(info->request.header.op, "deactivate")) {
         if(!DIDDocument_IsValid(doc))
             goto errorExit;
     }
 
```

Comment:
In my suggest,  we need to  merge these two condition statements from line. 80 to line. 86.

if  (strcmp(info->request.header.op, "deactivate")) {
    if (!doc || !DIDDocument_IsValid(doc))
         goto errorExit;
}